### PR TITLE
Financial Connections: for v3, added support for showing a v3 Data Access Notice + Legal Details Notice

### PR DESCRIPTION
**Pull request by kgaidis-stripe**
## Summary

This PR:
- Adds data and legal notices sheets that were removed [in another pr](https://github.com/stripe/stripe-ios/pull/3145)
- Fixes more SheetViewController bugs

Details to understand before reviewing PR:
- This is one PR, out of expected ~50+, to change the design of Financial Connections from a "V2" design to a "V3" design. [Here is a link to the designs](https://www.figma.com/file/wXQ8NJApqSJiLmxxANDRTs/%E2%9C%A8-Connections-3.0?type=design&node-id=2524-217093&mode=design&t=kXaBiAlgVsC7YcH1-0).
- The code here is not the finished version. It's just an iteration towards the final version. This code will be re-visited later. The goal is to go screen-by-screen and make easy changes to learn about all the difficulties in each screen. Later, all the learnings will be combined to do a "final" polish of all screens.
- The above also applies to design. The design here is not the finished version.
- This PR is merged into a feature branch `fc-v3`. It is not merged into `master`.
- There might be TODO's scattered throughout the code. That's expected and they will be fixed later. 

[Data Access Notice Model](https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/bank_connections/text/data/data_access_notice_v3.rb#L11)

[Legal Details Model](https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/bank_connections/text/data/legal_details_notice_v3.rb)

### Data Access Notice

Simulator Screenshot - iPhone 15 Pro - 2024-01-04 at 15 51 02

### Terms And Privacy Policy Notice

Simulator Screenshot - iPhone 15 Pro - 2024-01-04 at 15 51 08

### Video Of Links In Sheets Working (and can see animations)

https://github.com/stripe/stripe-ios/assets/105514761/ef5c1f41-495d-46b7-b9f6-160d8420834b

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/Models/FinancialConnectionsDataAccessNotice.swift [hunk 1]
```diff
@@ -8,11 +8,12 @@
 import Foundation
 
 struct FinancialConnectionsDataAccessNotice: Decodable {
+    let icon: FinancialConnectionsImage?
     let title: String
     let subtitle: String?
     let body: Body
     let connectedAccountNotice: String?
-    let learnMore: String
+    let disclaimer: String?
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/Models/FinancialConnectionsDataAccessNotice.swift:16]: [link to model info for double-checking](https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/bank_connections/text/data/data_access_notice_v3.rb#L5-15) (important they match backend)

File: StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/Models/FinancialConnectionsLegalDetailsNotice.swift [hunk 2]
```diff
@@ -8,12 +8,20 @@
 import Foundation
 
 struct FinancialConnectionsLegalDetailsNotice: Decodable {
+
+    let icon: FinancialConnectionsImage?
     let title: String
+    let subtitle: String?
     let body: Body
-    let learnMore: String
+    let disclaimer: String?
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/Models/FinancialConnectionsLegalDetailsNotice.swift:16]: [link to double check models](https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/bank_connections/text/data/legal_details_notice_v3.rb)

**Review by ccen-stripe — APPROVED** [review]
LGTM! is there plan to add screenshot tests for these new screens?

**Comment by kgaidis-stripe** [comment]
> is there plan to add screenshot tests for these new screens?

right now we don't have any, but if we have the time + the designs get settled its a good idea - right now there's too much changing

**Pull request merged**